File: YukariLauncher/src/main/java/net/kdt/pojavlaunch/customcontrols/mouse/DropGesture.java
package net.kdt.pojavlaunch.customcontrols.mouse;

import static org.lwjgl.glfw.CallbackBridge.sendKeyPress;

import android.os.Handler;

import com.arata.yukarilauncher.setting.AllStaticSettings;

import net.kdt.pojavlaunch.LwjglGlfwKeycode;

public class DropGesture implements Runnable{
    private final Handler mHandler;
    private boolean mActive;
/**
 * コンストラクタ。
 * このクラスの新しいインスタンスを初期化します。
 */
    public DropGesture(Handler mHandler) {
        this.mHandler = mHandler;
    }
/**
 * 「submit」メソッド。
 * このクラスに定義された機能メソッドです。
 */
    public void submit() {
        if(!mActive) {
            mActive = true;
            mHandler.postDelayed(this, AllStaticSettings.timeLongPressTrigger);
        }
    }
/**
 * 「cancel」メソッド。
 * このクラスに定義された機能メソッドです。
 */
    public void cancel() {
        mActive = false;
        mHandler.removeCallbacks(this);
    }
/**
 * 「run」メソッド。
 * このクラスに定義された機能メソッドです。
 */
    @Override
    public void run() {
        if(!mActive) return;
        sendKeyPress(LwjglGlfwKeycode.GLFW_KEY_Q);
        mHandler.postDelayed(this, 250);
    }
}

File: YukariLauncher/src/main/java/com/arata/yukarilauncher/ui/subassembly/customcontrols/mouse/DropGesture.kt
package com.arata.yukarilauncher.ui.subassembly.customcontrols.mouse

import android.os.Handler
import com.arata.yukarilauncher.setting.AllStaticSettings
import com.arata.yukarilauncher.utils.LwjglGlfwKeycode
import org.lwjgl.glfw.CallbackBridge

/** アイテムドロップの長押しジェスチャーを検出・処理するクラス。一定時間長押しでQキーを繰り返し送信します。 */
class DropGesture(private val mHandler: Handler) : Runnable {
    private var mActive = false

    /** ドロップジェスチャーを開始します。 */
    fun submit() {
        if (!mActive) {
            mActive = true
            mHandler.postDelayed(this, AllStaticSettings.timeLongPressTrigger.toLong())
        }
    }

    /** ドロップジェスチャーをキャンセルします。 */
    fun cancel() {
        mActive = false
        mHandler.removeCallbacks(this)
    }

    override fun run() {
        if (!mActive) return
        CallbackBridge.sendKeyPress(LwjglGlfwKeycode.GLFW_KEY_Q.toInt())
        mHandler.postDelayed(this, 250)
    }
}
